### PR TITLE
Add friendly JSONException catches

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Review by oliver-walker-hms — DISMISSED** [review]

File: src/com/hms_networks/americas/sc/config/ConfigFile.java [hunk 1]
```diff
@@ -99,15 +113,18 @@ public boolean fileExists() {
     return configFile.isFile();
   }
 
-
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/config/ConfigFile.java]: WS

File: src/com/hms_networks/americas/sc/config/ConfigFile.java [hunk 2]
```diff
@@ -57,15 +57,29 @@ public void read() throws ConfigFileReadException, ConfigFileParseException {
        * If unable to read from file, use default config and rethrow exception as a more
        * human-readable exception with an explanation.
        */
-      configurationObject = getDefaultConfigurationObject();
+      try {
+        configurationObject = getDefaultConfigurationObject();
+      } catch (JSONException e2) {
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/config/ConfigFile.java:62]: whats the significance of e2?

File: src/com/hms_networks/americas/sc/config/ConfigFile.java [hunk 3]
```diff
@@ -57,15 +57,29 @@ public void read() throws ConfigFileReadException, ConfigFileParseException {
...
+        throw new ConfigFileReadException(
+            "Failed to create a default configuration while handling an error with reading "
+                + "the configuration from file. Check the specified file path",
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/config/ConfigFile.java]:
```suggestion
                + "the configuration from file. Check the specified file path.",
```

File: src/com/hms_networks/americas/sc/config/ConfigFile.java (hunk 2 above)
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/config/ConfigFile.java:62]: I made a mistake in the last PR. This exception was not caught, and it was not declared as a thrown exception in the method signature.

File: src/com/hms_networks/americas/sc/config/ConfigFile.java (hunk 1 above)
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/config/ConfigFile.java]: removed

File: src/com/hms_networks/americas/sc/config/ConfigFile.java (hunk 3 above)
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/config/ConfigFile.java]: fixed

File: src/com/hms_networks/americas/sc/config/ConfigFile.java (hunk 2 above)
Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/config/ConfigFile.java:62]: I was commenting on the name "e2" but github just wasnt showing me the full scope and the fact that this is inside a catch with e already used.

**Review by TomKimsey — APPROVED** [review]

**Review by oliver-walker-hms — APPROVED** [review]

**Pull request merged**